contrib/install.sh

```diff
@@ -0,0 +1,1121 @@
+    4) interactive_cfilters ;;
+    5) interactive_ask_assume_valid ;;
+    6) interactive_assume_utreexo ;;
+    7) interactive_ssl ;;
+    8) interactive_ask_for_add_xpubs ;;
+    9) interactive_ask_for_add_descriptors ;;
+    R) interactive_review ;;
+    *)
+      clear
+      exit 0
+      ;;
+  esac
+}
+
+# func: interactive_select_network
+# 
+# Ask for which network the user want
+# to configure the floresta node
+interactive_select_network() {
+  choice=$(dialog --title "Floresta-Installer" \
+                  --menu "Which network do you want to use?" \
+                  11 45 25 \
+                  1 "bitcoin" \
+                  2 "signet" \
+                  3 "testnet" \
+                  4 "regtest" \
+                  3>&1 1>&2 2>&3)
+
+  case $choice in
+    1)
+      network="bitcoin"
+      interactive_setup
+      ;;
+    2)
+      network="signet"
+      interactive_setup
+      ;;
+    3)
+      network="testnet"
+      interactive_setup
+      ;;
+    4)
+      network="regtest"
+      interactive_setup
+      ;;
+    *) interactive_main_menu ;;
+  esac
+}
+
+# func: interactive_select_proxy
+#
+# This function setup a proxy if user wants add one
+interactive_select_proxy() {
+  local proxy_regex="^((http|https|socks5):\/\/)?((([0-9]{1,3}\.){3}[0-9]{1,3})|\[?[a-fA-F0-9:]+\]?):[0-9]{1,5}$"  # Matches optional protocol, IPv4, IPv6, and port
+
+  while true; do
+    # Prompt user for proxy input
+    proxy=$(dialog --title "Floresta-Installer (Set Proxy)" \
+                  --inputbox "Provide a proxy address (Optional: http://, https://, or socks5:// followed by IPv4, IPv6, or domain:port)" \
+                  10 60 \
+                  3>&1 1>&2 2>&3)
+
+    # Check if user pressed Cancel or ESC
+    if [ $? -ne 0 ]; then
+      dialog --title "Floresta-Installer" --msgbox "❌ Proxy input canceled. Returning to setup menu." 8 45
+      interactive_setup
+      return 1
+    fi
+
+    # Validate proxy format
+    if [[ "$proxy" =~ $proxy_regex ]]; then
+      dialog --title "Floresta-Installer" --msgbox "✅ Proxy set successfully: $proxy" 10 60
+      interactive_setup
+      return 0
+    else
+      dialog --title "Invalid Proxy" --msgbox "❌ The proxy address you entered is invalid.\n\nIt should be in the format:\n- http://IPv4:port\n- https://IPv6:port\n- socks5://IPv4:port\n- IPv4:port\n- [IPv6]:port\n\nExample: 192.168.1.100:8080 or [2001:db8::1]:9050\n\nReturning to setup menu." 12 60
+    fi
+  done
+}
+
+# func: interactive_connect
+#
+# This function setup a node to connect when daemon starts
+interactive_connect() {
+  local connect_regex="^((25[0-5]|2[0-4][0-9]|1[0-9]{2}|[1-9]?[0-9])\.){3}(25[0-5]|2[0-4][0-9]|1[0-9]{2}|[1-9]?[0-9])(:([0-9]{1,5}))?$"
+
+  while true; do
+    # Prompt user for proxy input
+    connect=$(dialog --title "Floresta-Installer (Set a node to connect)" \
+                  --inputbox "We'll connect ONLY to this node. It should be an ipv4 address in the format <address>[:<port>]." \
```

Comment:
On master, it can be an ipv4/6 and some dns address. Should we update this once we get a new release?